tests/unit/test_collector.py

```diff
@@ -586,7 +586,7 @@ def test_parse_links_presents_warning_on_html4_doctype(
     assert "pkg1-1.0" in parsed_links[0].url
     assert "pkg1-2.0" in parsed_links[1].url
 
-    assert len(caplog.records) == 1
+    assert len(caplog.records) == 0
```

Comment:
I'd strongly prefer to drop these tests.

Reply:
I'm happy to do that. But I'd still like to see a test that ensures that any validation we might put on indexes doesn't "leak" into how we handle `--find-links` (because `--find-links` is explicitly intended to support anything that dumps out an unstructured "bunch of links to files", such as a webserver auto-generated directory page). But I won't block somebody else merging this PR without that test, if people feel I'm being too pedantic.

Reply:
Ill add a test for `--find-links`.